data/functions.sql

```diff
@@ -214,7 +214,23 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql IMMUTABLE;
 
-CREATE OR REPLACE FUNCTION mz_calculate_road_level(highway_val text, railway_val text, aeroway_val text)
+CREATE OR REPLACE FUNCTION mz_calculate_ferry_level(way geometry)
+RETURNS SMALLINT AS $$
+DECLARE
+  way_length real := st_length(way);
+BEGIN
+  RETURN (
+    CASE
+      WHEN way_length > 51356 THEN  9
```

Comment:
I've changed the cutoffs in 3b66d2b to be based on the rendered size, 2px seemed like a decent level. This shows all the longer ferry routes at zoom 8 around SF, see comment below for updated screenshots.

The trouble with `ferry=primary` is that it's only set on about 100 objects at the moment, so really it's a non-standard niche tag. I don't think it's appropriate to support it or add it to OSM just so that we can render it; that's the worst kind of "tagging for the renderer". Vandalising OSM paired with a commit bothers _me_. If we want curated data with no "on the ground" basis, then we should be pulling that in from somewhere else or curating it ourselves.

Alternatively, we should think about what _physical_ attributes make a ferry route important, and I think it's closely related to "network importance", i.e: what does it do to the connectivity of the routing graph to remove it? How much extra delay, or how much of the route network is severed?